pkg/gardenlet/controller/shoot/shoot/reconciler_reconcile.go

```diff
@@ -567,22 +567,27 @@ func (r *Reconciler) runReconcileShootFlow(ctx context.Context, o *operation.Ope
 
 		syncPointAllSystemComponentsDeployed = flow.NewTaskIDs(
 			waitUntilNetworkIsReady,
+			waitUntilControlPlaneReady,
 			deployShootSystemResources,
 			deployCoreDNS,
 			deployNodeLocalDNS,
 			deployMetricsServer,
 			deployVPNShoot,
 			deployNodeProblemDetector,
 			deployKubeProxy,
-			deployManagedResourcesForAddons,
+			deployManagedResourcesForAddons, // includes apiserver-proxy
 			deployKubernetesDashboard,
 			deployNginxIngressAddon,
 		)
 
 		deployWorker = g.Add(flow.Task{
-			Name:         "Configuring shoot worker pools",
-			Fn:           flow.TaskFn(botanist.DeployWorker).RetryUntilTimeout(defaultInterval, defaultTimeout),
-			Dependencies: flow.NewTaskIDs(deployCloudProviderSecret, deployReferencedResources, waitUntilInfrastructureReady, initializeShootClients, waitUntilOperatingSystemConfigReady, waitUntilNetworkIsReady, createNewServiceAccountSecrets),
+			Name: "Configuring shoot worker pools",
+			Fn:   flow.TaskFn(botanist.DeployWorker).RetryUntilTimeout(defaultInterval, defaultTimeout),
+			Dependencies: flow.NewTaskIDs(
+				deployCloudProviderSecret, deployReferencedResources, waitUntilInfrastructureReady, initializeShootClients, waitUntilOperatingSystemConfigReady, createNewServiceAccountSecrets,
+				// wait for all node-critical components to be deployed
+				syncPointAllSystemComponentsDeployed,
```

Comment:
I don't understand this commit - the fact that the `ManagedResource`s were created does not mean that the resulting objects have been applied to the shoot cluster (GRM might be done or unavailable)?
In this light, Do we need this special handling at all?

Reply:
Yeah, you're kind of right.
If resource-manager is not available, it won't remove the taint 😄 But still, this is racy.

I'm fine with either a) removing the special handling b) introducing new logic to wait for the `ManagedResources` to be applied.
We could also keep the changed dependencies as they seem sensible to me independently from this change.

What's your vote?

Reply:
When a new node is created, GRM must be up so that it can approve the CSR for the kubelet's server certificate. If GRM was down, the kubelet would not start any pods anyways, right? Hence, aren't we "implicitly fine" here?

Reply:
Yeah, true. We should be fine for most cases. A potential race between the `ManagedResource` and `Node` controllers would only be a problem for newly created shoot clusters. 

This feature aims to improve the stability of existing clusters when new nodes are joined. It's probably not worth the effort as it can only be problematic during shoot creation, and is still very unlikely.
If we still face problems with this, we can improve it later on.

I will simply drop this commit :)